### ISSUE TITLE
fails most of the time

### ISSUE THREAD

**Issue by MaoHolden** [TudbuT/qft#16]
Hey Man, thank you for your quick fix of the server.

I tested several times, here are my results:

- Most of the times just one of the clients display connection successful but the other one doesn't display anything and just keep loading forever, sometimes is the sender, sometimes is the receiver.
- I was able to send a small image 2 times successfully.
- When the connection is successful in large files it disconnects after a few seconds, the sender keeps trying to resend packet and the receiver just do nothing.

Here are  two images from the third case:

![qft1](https://github.com/TudbuT/qft/assets/135930538/dfba3a3b-6925-4fbe-ac4b-a8cf3aa5f3b0)
![qft2](https://github.com/TudbuT/qft/assets/135930538/0c8ed0f4-9d1f-4e18-9c9b-e06859fc0802)

**Comment by MaoHolden** [comment]
Just want to mention that having a battle tested app like this would be awesome, not relying on servers to send files 🤯, please tell me if there's anything else I can do to help test, maybe run it with a higher debug level or something? Cheers!

**Comment by TudbuT** [comment]
Usually you can simply increase the delay or lower the bitrate when the connection fails.
In terms of the failed connection: It is expected that both sender and receiver connect within 10 seconds of eachother. If that isn't the case, it tends to fails sometimes.

**Comment by TudbuT** [comment]
> Just want to mention that having a battle tested app like this would be awesome, not relying on servers to send files 🤯, please tell me if there's anything else I can do to help test, maybe run it with a higher debug level or something? Cheers!

I have already sort of battle-tested it: I use it to distribute files of 20 and up to 100 gigabytes (a very large minecraft world download) frequently.

I haven't needed anyone else's help recently, but I'll be sure to contact you if/when I do. Thanks for the offer!

**Comment by MaoHolden** [comment]
This is a far as I gotten, a small % of a movie, maybe I'm talking nonsense, but is there a way to tell it to holepunch again after x time of no connection? or something like that

![t2-1](https://github.com/TudbuT/qft/assets/135930538/7d587644-ddde-43d5-87d5-0723b83fa0bc)
![Screenshot from 2023-11-08 12-46-23](https://github.com/TudbuT/qft/assets/135930538/539d29da-1b36-4742-aad3-9256c19f6b44)

**Comment by MaoHolden** [comment]
Also I wouldn't know the value for "Resume from" to restart the download when it gets stuck

**Comment by MaoHolden** [comment]
What I noticed is that once it says "10s passed since last package" that's it, it completely stuck.

**Comment by TudbuT** [comment]
Yeah this is one of the cases where your internet connection is either too slow or too unreliable. How fast is your internet? Are you using WiFi?

> but is there a way to tell it to holepunch again after x time of no connection
I could probably make it do that actually, but I'm not sure how I'd implement it at this moment. I'll have a think about that later.

**Comment by MaoHolden** [comment]
Yup I'm trying to send a movie from one computer using WIFI to another computer also using WIFI but with a Router/VPN

I also tried to send it to a friend using starlink and couldn't make it work, I'm gonna try again sending to Starlink without the VPN, that could be the reason.

Just wondering, how does torrents re-stablish connection after they "disconnect", because in the computer with VPN I can use torrents perfectly.

![photo1699556169](https://github.com/TudbuT/qft/assets/135930538/e0255cb6-5ec2-4063-a639-7b457e9d3c86)

**Comment by MaoHolden** [comment]
I had better results transfering to a Startlink connection without the VPN, I'm gonna try tomorrow with a 5GB file and report back. This could be really useful for me since we have a ton of videos, that's why I'm so interested. And bittorrent sync went private :'( 

Would be awesome if it could reconnect automatically after losing connection for certain amount of time. Do you rekon that would be too hard? I would like to suggest it as a feature. 😸

**Comment by TudbuT** [comment]
Should be pretty doable to add a way to make it reconnect. I'll take a look.

**Comment by MaoHolden** [comment]
Thanks, since it looks like my connections are not pretty stable enable automatic reconnection should pretty much solve my problem
